test/test_unistd.rs

```diff
@@ -576,6 +577,39 @@ fn test_symlinkat() {
     );
 }
 
+
+#[test]
+fn test_unlinkat() {
+    let tempdir = tempfile::tempdir().unwrap();
+    let filename = "foo.txt";
+    let dirname = "foo_dir";
+    let filepath = tempdir.path().join(filename);
+    let dirpath = tempdir.path().join(dirname);
+
+    // Create file
+    File::create(&filepath).unwrap();
+
+    // Create dir
+    DirBuilder::new().recursive(true).create(&dirpath).unwrap();
+
+    // Get file descriptor for base directory
+    let dirfd = fcntl::open(tempdir.path(), fcntl::OFlag::empty(), stat::Mode::empty()).unwrap();
+
+    // Attempt unlink dir at relative path without proper flag
+    panic::set_hook(Box::new(|_info| {}));
+    let result = std::panic::catch_unwind(|| unlinkat(Some(dirfd), dirname, UnlinkatFlags::NoRemoveDir).unwrap());
+    assert!(result.is_err());
+    assert!(dirpath.exists());
+
+    // Attempt unlink dir at relative path with proper flag
+    unlinkat(Some(dirfd), dirname, UnlinkatFlags::RemoveDir).unwrap();
+    assert!(!dirpath.exists());
+
+    // Attempt unlink file at relative path
+    unlinkat(Some(dirfd), filename, UnlinkatFlags::NoRemoveDir).unwrap();
```

Comment:
A good unit test should do only the minimum.  You should break this test up into 3: the panic test, the successful remove file test, and the successful remove dir test.